### DISCUSSION TITLE
tvOS not logging on real device

### DISCUSSION THREAD

**Discussion by mmmcheese** [kean/Pulse#168] [Q&A] [answered]
I'm developing an app for both iOS and tvOS, and when I run it on simulators, everything works fine. When I try to run it on devices, the iPad I'm using produces logs and also connects remotely. On tvOS, I just get a perpetual No Recorded Logs, and an error `Failed to load info. The file "logs.sqlite" couldn't be opened because there is no such file.`

![Screen Shot 2023-03-01 at 9 27 37 AM](https://user-images.githubusercontent.com/65556895/222216773-458b507e-5403-4361-8f6b-f9281e26e7cc.jpeg)

**Comment by kean** [comment]
I would appreciate some help debugging it as I don't have an Apple TV device and only tested in a simulator. @mmmcheese, it's likely you are the first user who ever attempted running on Apple TV 😊

**Reply by mmmcheese** [reply]
Oh no. What do you need me to do?

**Reply by kean** [reply]
It would, of course, be ideal if you could create a PR with a fix. Because I'm not going to be able to test it properly. I'm thinking it's something that has to do with permissions / file paths. I'm not sure what limitations tvOS has in that area.

**Reply by mmmcheese** [reply]
I honestly am not even sure what the problem is to begin with. I'm not seeing any other errors. When I load the console, this is all I see. The same Console screen on the simulator of course has no issues.

![Screen Shot 2023-03-01 at 2 25 44 PM](https://user-images.githubusercontent.com/65556895/222279547-297259aa-e644-4422-ae8c-63bb138a89e3.jpeg)
![Screen Shot 2023-03-01 at 2 29 13 PM](https://user-images.githubusercontent.com/65556895/222280115-e2571c54-b81e-4cb3-827a-588efa281abe.jpeg)

A bunch of options are missing as well.

**Reply by mmmcheese** [reply]
From what I've read "There's an inconsistency regarding the Documents folder: you can create files in it when you run your app in the Simulator. But on the actual Apple TV, it just fails."

Looks like the only storage available on tvOS is the temp cache directory. hopefully that helps.

**Reply by mmmcheese** [reply]
I have a fix for this. I will hopefully be providing a PR shortly.

**Reply by mmmcheese** [reply]
PR has been posted.

**Reply by mmmcheese** [reply]
@kean So I know it's been a while since you pushed the changes to the main branch, but it appears as though they don't work. I have not investigated why the changes I provided work, and the ones you pushed to the main branches don't but unfortunately the main branch still doesn't work on tvOS hardware. In the meantime, I'm using my fork for my own project until we can figure out a solution.

**Reply by kean** [reply, marked as answer]
Thanks for reporting it. I think I know what the issue is. Please keep an eye on the new release. I'll push 3.6.1 soon with a fix.

**Reply by mmmcheese** [reply]
That fixed it. Thanks.
